modules/fedimint-dummy-client/src/db.rs

```diff
@@ -70,69 +69,26 @@ pub async fn migrate_to_v1(
-    let mut new_active_states = Vec::new();
-    for (active_state, _) in active_states {
-        // Try to decode the bytes as a `DynState`
-        let dynstate = DynState::from_bytes(active_state.as_slice(), &decoders)?;
-        let typed_state = dynstate
-            .as_any()
-            .downcast_ref::<DummyStateMachine>()
-            .expect("Unexpected DynState suppilied to migration function");
-
-        match typed_state {
-            DummyStateMachine::Unreachable(_, _) => {
-                // Try to parse the bytes as the `Unreachable` struct to simulate a deleted
-                // state. In a real migration, `DynState::from_bytes` will
-                // fail since `DummyStateMachine::Unreachable` will not exist.
-                if let Ok(unreachable) =
-                    Unreachable::consensus_decode_vec(active_state.clone(), &decoders)
-                {
-                    new_active_states.push(
-                        DummyStateMachine::OutputDone(unreachable.amount, unreachable.operation_id)
-                            .into_dyn(module_instance_id),
-                    );
-                }
-            }
-            state => new_active_states.push(state.clone().into_dyn(module_instance_id)),
-        }
-    }
-
-    let mut new_inactive_states = Vec::new();
-    for (inactive_state, _) in inactive_states {
-        // Try to decode the bytes as a `DynState`
-        let dynstate = DynState::from_bytes(inactive_state.as_slice(), &decoders)?;
-        let typed_state = dynstate
-            .as_any()
-            .downcast_ref::<DummyStateMachine>()
-            .expect("Unexpected DynState suppilied to migration function");
-
-        match typed_state {
-            DummyStateMachine::Unreachable(_, _) => {
-                // Try to parse the bytes as the `Unreachable` struct to simulate a deleted
-                // state. In a real migration, `DynState::from_bytes` will
-                // fail since `DummyStateMachine::Unreachable` will not exist.
-                if let Ok(unreachable) =
-                    Unreachable::consensus_decode_vec(inactive_state.clone(), &decoders)
-                {
-                    new_inactive_states.push(
-                        DummyStateMachine::OutputDone(unreachable.amount, unreachable.operation_id)
-                            .into_dyn(module_instance_id),
-                    );
-                }
-            }
-            state => new_inactive_states.push(state.clone().into_dyn(module_instance_id)),
-        }
+/// Maps all `Unreachable` states in the state machine to `OutputDone`
+pub(crate) fn get_v1_migrated_state(
+    bytes: &[u8],
+    operation_id: OperationId,
+) -> anyhow::Result<Option<(Vec<u8>, OperationId)>> {
+    let decoders = ModuleDecoderRegistry::default();
+    let mut cursor = std::io::Cursor::new(bytes);
+    let module_instance_id =
+        fedimint_core::core::ModuleInstanceId::consensus_decode(&mut cursor, &decoders)?;
```

Comment:
More so - modules should never see their instance_id. We break that in couple of places of our API, but I plan to address it at some point.

It enforces hygiene and architecture that will be important long term.

Reply:
Sure, should be an easy fix

Reply:
This migration would need to be applied to all modules though, since all modules have the `module_instance_id` embedded in the `DynState`. The dummy client has a separate issue that we duplicate the `module_instance_id` in the state as well.